src/ansi-c/ansi_c_entry_point.cpp

```diff
@@ -473,7 +474,19 @@ bool generate_ansi_c_start_function(
       }
     }
     else
-      UNREACHABLE;
+    {
+      const namespacet ns{symbol_table};
+      const std::string main_signature = type2c(symbol.type, ns);
+      throw invalid_source_file_exceptiont{
+        "'main' with signature '" + main_signature +
+        "' found,"
+        " but expecting one of:\n"
+        "   int main(void)\n"
+        "   int main(int argc, char *argv[])\n"
+        "   int main(int argc, char *argv[], char *envp[])\n"
+        "If this is a non-standard main entry point please provide a custom\n"
+        "entry function and point to it via cbmc --function instead"};
```

Comment:
This is a great error message :green_heart: 
:grey_question: If I do `cbmc main.c --function main` does this work? No worries if not, but if it does, I'd be half tempted to give that as the advice

Reply:
It doesn’t unfortunately. What does work is adding something like

```
void harness(void) {
  main(NULL);
}
```

and pointing at that.